File: python/7 kyu/420.py
# https://www.codewars.com/kata/59b0ab12cf3395ef68000081/train/python

# Passed

def to_12_hour_time(time_string):
    hours, minutes = int(time_string[:2]), int(time_string[2:])
    is_am = not hours >= 12
    if hours == 0:
        hours = 12
    elif hours >= 13:
        hours -= 12
    
    result = f"{hours}:{str(minutes).zfill(2)} {'am' if is_am else 'pm'}"
    return result

output = to_12_hour_time('2145')
print(output)

File: nodejs/7 kyu/420.js
// https://www.codewars.com/kata/599da159a30addffd00000af/train/javascript

// Passed

function collision(x1, y1, radius1, x2, y2, radius2) {
  let getDistance = (x1, y1, x2, y2) =>
    ((x1 - x2) ** 2 + (y1 - y2) ** 2) ** 0.5;
  let result = radius1 + radius2 >= getDistance(x1, y1, x2, y2);
  return result;
}

const output = collision(-1, 1, 10, -10.1, 1.1, 1);
console.log(output);